src/client/datascience/editor-integration/codewatcher.ts

```diff
@@ -115,11 +111,13 @@ export class CodeWatcher implements ICodeWatcher {
         return this.version;
     }
 
-    public getCachedSettings(): IJupyterSettings | undefined {
-        return this.cachedSettings;
-    }
-
     public getCodeLenses() {
+        const document = this.document;
+        if (document && document.version != this.version) {
```

Comment:
So I think that this will be a slight regression in that if you change the cell marker regex in the settings, it won't generate new code lenses until you modify a document.

Reply:
I think we can trigger the change in code lenses when we detect a change in this setting, this way VS Code will re-query code lenses and we'll provide the new ones.

Reply:
tested by changing the settings in another window, still updates in around a second.
Making a change to the settings fires off a codeLens change event